Hearthstone Deck Tracker/FlyoutControls/Options/Tracker/TrackerPlugins.xaml.cs

```diff
@@ -65,52 +70,319 @@ private void ButtonOpenPluginsFolder_OnClick(object sender, RoutedEventArgs e)
 						{
-							var path = Path.Combine(dir, Path.GetFileNameWithoutExtension(pluginPath));
-							ZipFile.ExtractToDirectory(pluginPath, path);
-							if (Directory.GetDirectories(path).Length == 1 && Directory.GetFiles(path).Length == 0)
-							{
-								Directory.Delete(path, true);
-								ZipFile.ExtractToDirectory(pluginPath, dir);
-							}
-							plugins++;
+							Directory.Delete(path, true);
+							ZipFile.ExtractToDirectory(pluginPath, dir);
 						}
+						plugins++;
 					}
-					if(plugins <= 0) 
-						return;
-					var result = await Core.MainWindow.ShowMessageAsync("Plugins installed",
-						$"Successfully installed {plugins} plugin(s). \n Restart now to take effect?", MessageDialogStyle.AffirmativeAndNegative);
-
-					if(result != MessageDialogResult.Affirmative)
-						return;
-					Core.MainWindow.Restart();
 				}
+				if(plugins <= 0)
+					return;
+				var result = await Core.MainWindow.ShowMessageAsync("Plugins installed",
+					$"Successfully installed {plugins} plugin(s). \n Restart now to take effect?", MessageDialogStyle.AffirmativeAndNegative);
+
+				if(result != MessageDialogResult.Affirmative)
+					return;
+				Core.MainWindow.Restart();
 			}
 			catch(Exception ex)
 			{
 				Log.Error(ex);
 				Core.MainWindow.ShowMessage("Error Importing Plugin", $"Please import manually to {dir}.").Forget();
 			}
 		}
+
+		#endregion
+
+
+
+		#region JSON functions
+
+		private List<Plugin> GetPlugins()
+		{
+			//Create list of repo URLS to compare to.
+			var repos = new List<string>();
+			foreach (var p in ListBoxPlugins.Items)
+			{
+				var plugin = p as PluginWrapper;
+				repos.Add(releaseUrl(plugin?.Repourl));
+			}
+			Log.Info("downloading available plugin list.");
+			var pluginList = new List<Plugin>();
+			var wc = new WebClient();
+			wc.Headers["User-Agent"] = $"Hearthstone Deck Tracker {Core.Version} @ Hearthsim";
+			var json = JObject.Parse(wc.DownloadString("https://raw.githubusercontent.com/HearthSim/HDT-Plugins/master/plugins.json"));
+			foreach(var plugin in json["data"])
+			{
+				var baseUrl = plugin["url"].ToString();
+				var releaseUrl = baseUrl + "/releases/latest";
+				if(repos.Contains(releaseUrl))
+					continue;
+				var Plugin = new Plugin
+				{
+					Author = plugin["author"].ToString(),
+					Description = plugin["description"].ToString(),
+					Name = plugin["title"].ToString(),
+					ReleaseUrl = releaseUrl
+				};
+				pluginList.Add(Plugin);
+			}
+			wc.Dispose();
+			return pluginList;
+		}
+
+		private List<Plugin> GetUpdates()
+		{
+			try
+			{
+				Log.Info("Checking for updates to installed plugins.");
+				var updateList = new List<Plugin>();
+				if(ListBoxPlugins.Items.IsEmpty)
+					return null;
+				var wc = new WebClient();
+				foreach(var p in ListBoxPlugins.Items)
+				{
+					var plugin = p as PluginWrapper;
+					wc.Headers["User-Agent"] = $"Hearthstone Deck Tracker {Core.Version} @ Hearthsim";
```

Comment:
The header is inside the loop because when declared outside the loop, it would be reset to null after the first iteration and GitHub would not serve the request.